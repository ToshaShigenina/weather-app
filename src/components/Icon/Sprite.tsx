import React from 'react';
import { createPortal } from 'react-dom';
import { ReactComponent as SpriteSvg } from './svg/sprite.svg'

import './css/sprite.css';

const portalNode = document.createElement('div');
portalNode.setAttribute('id', 'sprite');
portalNode.setAttribute('data-testid', 'sprite');
portalNode.hidden = true;

const Sprite: React.FC<React.SVGAttributes<SVGElement>> = (props) => {
	React.useEffect(() => {
		console.log(portalNode)
		document.body.appendChild(portalNode);
		return () => portalNode.remove();
	});
	return createPortal(<SpriteSvg { ...props } />, portalNode)
};

export default Sprite;