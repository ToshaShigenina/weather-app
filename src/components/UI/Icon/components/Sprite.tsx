import React, { FC, SVGAttributes, useEffect } from 'react';
import { createPortal } from 'react-dom';
import { ReactComponent as SpriteSvg } from '../svg/sprite.svg'

import '../css/sprite.css';

const portalNode = document.createElement('div');
portalNode.setAttribute('id', 'sprite');
portalNode.setAttribute('data-testid', 'sprite');
portalNode.hidden = true;

export const Sprite: FC<SVGAttributes<SVGElement>> = (props) => {
	useEffect(() => {
		document.body.appendChild(portalNode);
		return () => portalNode.remove();
	});
	return createPortal(<SpriteSvg { ...props } />, portalNode)
};
