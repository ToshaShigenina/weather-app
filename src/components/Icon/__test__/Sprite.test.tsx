import React from 'react';
import { mountTest } from '../../../tests';

import Sprite from '../Sprite';

describe('Sprite', () => {
	mountTest(Sprite);
});