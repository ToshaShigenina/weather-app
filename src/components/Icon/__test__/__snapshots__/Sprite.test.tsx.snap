// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sprite спрайт добавляется в боди 1`] = `
<body>
  <div />
  <div
    data-testid="sprite"
    hidden=""
    id="sprite"
  >
    <svg>
      sprite.svg
    </svg>
  </div>
</body>
`;
